src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/DeclarationTests.cs

```diff
@@ -2139,5 +2144,59 @@ .maxstack  4
 }");
             });
         }
+
+        [Fact]
+        public void DuplicateDeclaration()
+        {
+            var source =
+@"class C
+{
+    static void M()
+    {
+        var x = 0;
+#line 999
+    }
+}";
+            var compilation0 = CreateStandardCompilation(source, options: TestOptions.DebugDll);
+
+            WithRuntimeInstance(compilation0, runtime =>
+            {
+                var context = CreateMethodContext(runtime, "C.M", atLineNumber: 999);
+
+                DkmClrCompilationResultFlags flags;
+                CompilationTestData testData;
+                string error;
+                CompileDeclaration(context, "var x = 1;", out flags, out testData, out error);
+                Assert.Equal("error CS0136: A local or parameter named 'x' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter", error);
+            });     
+        }
+
+        [Fact]
+        public void DuplicateDeclarationInOutVar()
+        {
+            var source =
+@"class C
+{
+    static void F(out double x, out int y) => x = y = 4;
+
+    static void M()
+    {
+#line 999
+        F(out var x, out var y);
+    }
+}";
+            var compilation0 = CreateStandardCompilation(source, options: TestOptions.DebugDll);
+
+            WithRuntimeInstance(compilation0, runtime =>
+            {
+                var context = CreateMethodContext(runtime, "C.M", atLineNumber: 999);
```

Comment:
`#line 999` shouldn't be necessary for either this test or the previous test. In both cases, the existing locals are in scope.